### ISSUE TITLE
[Checkbox] Add indeterminate state

### ISSUE THREAD

**Issue by lifeart** [Semantic-Org/Semantic-UI#2440]
Checkbox has 3 states, one missed
https://developer.mozilla.org/en-US/docs/Web/CSS/%3Aindeterminate

**Comment by nmschulte-aviture** [comment]
I just noticed this as well.  This seems a simple and valuable feature.

It's worth noting that the `indeterminate` property has no associated HTML attribute (and thus is only mutable via e.g. JavaScript), and doesn't actually enable tri-state features on a checkbox.

**Comment by jlukic** [comment]
I am not familiar with its use. Could someone help explain the use of this property for developers? Is it something manually set? Or is this the state on an 'untouched' form?

**Comment by nmschulte-aviture** [comment]
It is an HTML property introduced with HTML5.  It is purely a visual effect, causing browsers with the `indeterminate` property to render an "indeterminate" style (typically a dash, as opposed to a checkmark).

As there is no associated HTML attribute with the property, there is no way to set it via HTML.  It can only be set programmatically.  Further, as there is no HTML attribute, there is no way to leverage it in CSS selectors.

Basically, the property is useless as far as a request to Semantic UI goes.  Instead, this request is really "support an indeterminate state/style for checkboxes, akin to HTML5's `indeterminate` property."  S-UI will need to use a named style to apply any styling for indeterminate states, and thus the only use of the `indeterminate` property in its behavior is for compatabilities sake.

**Comment by lifeart** [comment]
indeterminate property usful for any Rule setter (allow, disallow, inherit)
And indeterminate used as "inherit" visual state.

**Comment by jlukic** [comment]
Oh totally forgot about this litter bugger.

**Comment by nmschulte-aviture** [comment]
@lifeart Ah!  Thanks for the tip about the pseudo-class (:indeterminate)!  My comment about not being able to leverage it in CSS is wrong.

**Comment by lifeart** [comment]
```

// if we call it as onclick event ex: onclick: setCheckState(this,event);
// checkbox states change continuously

// if we call function as setCheckState(domEl,2,domLabel);
// checkbox state will set as 2nd param value

var setCheckState = function (el,state,label) {

	var attrName = 'data-state';
	var currentState = null;
	var stateText = '';

	// if this is an event-like call
	if (typeof state === 'object') {

		state.stopPropagation();

		label = App.labelsMap[el.id];
		currentState = parseInt(el.getAttribute(attrName));


		if (el.readOnly) {
			el.checked=el.readOnly=false;
			state = 0;
			stateText = SAY('Disallow'); //Disallow
		} else if (!el.checked) {
			el.readOnly=el.indeterminate=true;
			state = 2;
			stateText = SAY('Inherit'); //Inherit
		} else {
			state = 1;
			stateText = SAY('Allow'); //Allow
		}
		if (label) {
			label.innerHTML = stateText;
		}

		el.setAttribute(attrName,state);

	} else {

		switch (state) {

			case 'unchecked':
			case 'false':
			case '0':
			case false:
			case 0:
				el.checked=el.readOnly=el.indeterminate=false;
				stateText = SAY('Disallow'); //Disallow
				state = 0;
				break;
			case 'checked':
			case 'true':
			case '1':
			case 1:
			case true:
				el.checked=true;
				el.readOnly=el.indeterminate=false;
				stateText =  SAY('Allow'); //Allow
				state = 1;
				break;
			case 'indeterminate':
			case '2':
			case 2:
			case undefined:
				el.checked=false;
				el.readOnly=el.indeterminate=true;
				stateText = SAY('Inherit');
				state = 2;
				break;
			default:
				break;

		}

		el.setAttribute(attrName,state);

		if (label) {
			label.innerHTML = stateText;
		}

	}

}
```

**Comment by jlukic** [comment]
Watch this.

**Comment by jlukic** [comment]
I wish there was a verb in the english language for "to make something indeterminate".

**Comment by jlukic** [comment]
Added updates, along with a few new examples. Grouped checkboxes for now require you to add your own application logic. Would make sense to do something like allowing users to specify a  `parent` checkbox, but I just don't have enough time. 
http://beta.semantic-ui.com/modules/checkbox.html#grouped-checkboxes

* Redid icon font to only use necessary characters
* Added a bunch more theming variables
* Added indeterminate state
* Added new behaviors for setting state without callbacks
